app/controllers/course/user_notifications_controller.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+class Course::UserNotificationsController < Course::ComponentController
+  def mark_as_read
+    notification = current_user.notifications.find(params[:user_notification_id])
```

Comment:
Suggest to define this route on member in route.rb, then do a load and authorised resource for permissions.

Reply:
Fixed!